### PR TITLE
[Draft][Do not merge]Convert sidebar from paper-listbox to mwc-list

### PR DESCRIPTION
**Pull request by donkawechico**
## Proposed change

Change sidebar to use `mwc-list` instead of `paper-listbox`.

This fixes an issue with quick bar where shortcuts won't work if you click on the left nav at all (because iron behaviors have a bug where it stops propagation of keyboard events: https://github.com/PolymerElements/iron-menu-behavior/issues/56)

sidebar_mwclist_1

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [X] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by donkawechico** [comment]
### Needed before merging
* Need some folks to test for feature parity:
  - Editing / Reordering / Hiding
  - Collapsed / Narrow views
  - Tooltips
  - Navigation
  - Other?
* Stylings do not yet match existing styles. Should they? Or should we keep consistent with as many mwc-list defaults as possible
* Find and remove unnecessary code/stylings that were there to work around `paper-listbox` behaviors

File: src/components/ha-sidebar.ts [hunk 1]
```diff
@@ -273,15 +275,16 @@ class HaSidebar extends LitElement {
...
+        <mwc-list-item class="spacer" disabled></mwc-list-item>
         ${this.editMode && this._hiddenPanels.length
           ? html`
               ${this._hiddenPanels.map((url) => {
                 const panel = this.hass.panels[url];
                 if (!panel) {
                   return "";
                 }
-                return html`<paper-icon-item
+                return html`<mwc-list-item
```
Comment (zsarnett) [inline src/components/ha-sidebar.ts:287]: Should use `graphic="icon"` here

Comment (zsarnett) [reply src/components/ha-sidebar.ts:287]: Then update the slot of the item to `slot="graphic`

File: src/components/ha-sidebar.ts [hunk 2]
```diff
@@ -292,106 +295,92 @@ class HaSidebar extends LitElement {
...
+                    "ui.sidebar.external_app_configuration"
+                  )}
+                  href="#external-app-configuration"
+                  tabindex="-1"
+                  @click=${this._handleExternalAppConfiguration}
+                  @mouseenter=${this._itemMouseEnter}
+                  @mouseleave=${this._itemMouseLeave}
+                >
                   <ha-svg-icon
                     slot="item-icon"
```
Comment (zsarnett) [inline src/components/ha-sidebar.ts:325]:
```suggestion
                    slot="graphic"
```

File: src/components/ha-sidebar.ts [hunk 3]
```diff
@@ -292,106 +295,92 @@ class HaSidebar extends LitElement {
...
-                  "ui.sidebar.external_app_configuration"
-                )}
-                href="#external-app-configuration"
-                tabindex="-1"
-                @click=${this._handleExternalAppConfiguration}
-                @mouseenter=${this._itemMouseEnter}
-                @mouseleave=${this._itemMouseLeave}
-              >
-                <paper-icon-item>
+              <mwc-list-item>
```
Comment (zsarnett) [inline src/components/ha-sidebar.ts:312]: Should use `graphic="icon"` here

Comment (zsarnett) [reply src/components/ha-sidebar.ts:312]:
```suggestion
              <mwc-list-item
```

File: src/components/ha-sidebar.ts [hunk 4]
```diff
@@ -292,106 +295,92 @@ class HaSidebar extends LitElement {
...
-              `
-            : ""}
-        </paper-icon-item>
-      </div>
-
-      <a
+        <ha-svg-icon slot="graphic" .path=${mdiBell}></ha-svg-icon>
+        ${!this.expanded && notificationCount > 0
+          ? html`
+              <span class="notification-badge" slot="graphic">
```
Comment (zsarnett) [inline src/components/ha-sidebar.ts:350]:
```suggestion
              <span class="notification-badge">
```

Comment (zsarnett) [reply src/components/ha-sidebar.ts:350]: Does this work? not sure

File: src/components/ha-sidebar.ts [hunk 5]
```diff
@@ -292,106 +295,92 @@ class HaSidebar extends LitElement {
...
+                ${notificationCount}
+              </span>
+            `
+          : ""}
+        ${hass.localize("ui.notification_drawer.title")}
+        ${this.expanded && notificationCount > 0
+          ? html` <span class="notification-badge">${notificationCount}</span> `
```
Comment (zsarnett) [inline src/components/ha-sidebar.ts:357]:
```suggestion
          ? html`<span class="notification-badge">${notificationCount}</span>`
```

File: src/components/ha-sidebar.ts [hunk 6]
```diff
@@ -292,106 +295,92 @@ class HaSidebar extends LitElement {
...
-      </a>
-      <div disabled class="bottom-spacer"></div>
+        <ha-user-badge
+          .user=${hass.user}
+          .hass=${hass}
+          slot="graphic"
+        ></ha-user-badge>
+        ${hass.user ? hass.user.name : ""}
+      </mwc-list-item>
+      <mwc-list-item disabled class="bottom-spacer"></mwc-list-item>
```
Comment (zsarnett) [inline src/components/ha-sidebar.ts:383]: https://github.com/material-components/material-components-web-components/tree/master/packages/list#dividers

If you use a divider does it produce the same result?

File: src/components/ha-sidebar.ts [hunk 7]
```diff
@@ -255,12 +258,11 @@ class HaSidebar extends LitElement {
             : "Home Assistant"}
         </div>
       </div>
-      <paper-listbox
-        attr-for-selected="data-panel"
+      <mwc-list
+        activatable
+        @selected=${this._selectItem}
         class="ha-scrollbar"
-        .selected=${hass.panelUrl}
```
Comment (bramkragten) [inline src/components/ha-sidebar.ts:261]: You should set `activated` on the panel that is active

File: src/components/ha-sidebar.ts [hunk 8]
```diff
@@ -292,106 +295,92 @@ class HaSidebar extends LitElement {
...
-                )}
-                href="#external-app-configuration"
-                tabindex="-1"
-                @click=${this._handleExternalAppConfiguration}
-                @mouseenter=${this._itemMouseEnter}
-                @mouseleave=${this._itemMouseLeave}
-              >
-                <paper-icon-item>
+              <mwc-list-item>
+                <a
```
Comment (bramkragten) [inline src/components/ha-sidebar.ts:313]: The anchor should be outside of the item, you can no longer right click and open in a new tab now. Also keyboard navigation doesn't work now.

File: src/components/ha-sidebar.ts [hunk 9]
```diff
@@ -292,106 +295,92 @@ class HaSidebar extends LitElement {
...
-          </span>
-          ${this.expanded && notificationCount > 0
-            ? html`
-                <span class="notification-badge">${notificationCount}</span>
-              `
-            : ""}
-        </paper-icon-item>
-      </div>
-
-      <a
```
Comment (bramkragten) [inline src/components/ha-sidebar.ts:368]: The anchor should not be removed, it removes functionality

File: src/components/ha-sidebar.ts (hunk 6 above)
Comment (donkawechico) [reply src/components/ha-sidebar.ts:383]: I saw that, but I assumed that dividers were just the little `<hr />` lines inbetween list items. I'll play with that.

File: src/components/ha-sidebar.ts (hunk 7 above)
Comment (donkawechico) [reply src/components/ha-sidebar.ts:261]: It seems to already do that by default.

image

**Comment by donkawechico** [comment]
Closing this PR as the overall approach has changed radically. I've put up a PR for my own reference and so I can have early ad-hoc discussions with other devs: https://github.com/home-assistant/frontend/pull/7573. At the time of this writing, it is not worth reviewing as it's full of junk and unused code.

**Pull request closed without merging**